### PR TITLE
BatchAccountIdentity constructor breaking change

### PR DESCRIPTION
**Pull request by paterasMSFT**
The previous release introduced a breaking type ambiguity to BatchAccountIdentity's constructor. This PR corrects that.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -12,11 +12,11 @@
   -->
   <ItemGroup Condition="'$(IsClientLibrary)' != 'true'">
     <PackageReference Update="Microsoft.Azure.Amqp" Version="2.4.11" />
-    <PackageReference Update="Microsoft.Azure.Batch" Version="15.0.0" />
+    <PackageReference Update="Microsoft.Azure.Batch" Version="15.1.0" />
     <PackageReference Update="Microsoft.Azure.Devices.Client" Version="1.23.2" />
     <PackageReference Update="Microsoft.Azure.Devices" Version="1.19.0" />
     <PackageReference Update="Microsoft.Azure.KeyVault.Core" Version="3.0.3" />
-    <PackageReference Update="Microsoft.Azure.Management.Batch" Version="4.2.0" />
+    <PackageReference Update="Microsoft.Azure.Management.Batch" Version="14.1.0" />
```
Comment (wiboris) [inline eng/Packages.Data.props:19]: 14.1.0 or 15.1.0?

Comment (paterasMSFT) [reply eng/Packages.Data.props:19]: [14.1](https://github.com/Azure/azure-sdk-for-net/blob/f6710ed0f9327aa2499f04d62d5867c89c162253/sdk/batch/Microsoft.Azure.Management.Batch/CHANGELOG.md). The ControlPlane and DataPlane have different versions.

**Review by wiboris — APPROVED** [review]

File: sdk/batch/Microsoft.Azure.Batch.Conventions.Files/src/Microsoft.Azure.Batch.Conventions.Files.csproj [hunk 2]
```diff
@@ -12,15 +12,14 @@
...
-    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
     <SignAssembly>true</SignAssembly>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="System.Text.Encodings.Web" />
-    <PackageReference Include="Microsoft.Azure.Batch" />
+    <PackageReference Include="Microsoft.Azure.Batch" Version="15.1.0" />
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Batch.Conventions.Files/src/Microsoft.Azure.Batch.Conventions.Files.csproj:22]: This shouldn't be needed - it should pick up the version from the Packages.Data.Props.

This actually shows up as an error in Visual Studio:

```
Severity	Code	Description	Project	File	Line	Suppression State
Error		The package reference 'Microsoft.Azure.Batch' should not specify a version.  Please specify the version in 'E:\development\microsoft\public\azure-sdk-for-net\eng\Packages.Data.props' or set VersionOverride to override the centrally defined version.	Microsoft.Azure.Batch.FileStaging	E:\development\microsoft\public\azure-sdk-for-net\sdk\batch\Microsoft.Azure.Batch.FileStaging\src\Microsoft.Azure.Batch.FileStaging.csproj	1	
```

File: sdk/batch/Microsoft.Azure.Batch.FileStaging/src/Microsoft.Azure.Batch.FileStaging.csproj [hunk 3]
```diff
@@ -18,13 +18,12 @@
...
   <PropertyGroup>
     <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
-    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="System.Text.Encodings.Web" />
     <PackageReference Include="WindowsAzure.Storage" />
-    <PackageReference Include="Microsoft.Azure.Batch" />
+    <PackageReference Include="Microsoft.Azure.Batch" Version="15.1.0" />
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Batch.FileStaging/src/Microsoft.Azure.Batch.FileStaging.csproj:26]: See the other comment on hard coding versions outside of Packages.Data.Props

File: sdk/batch/Microsoft.Azure.Management.Batch/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Release History
 
+## 14.1.0 (2021-09-01)
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Management.Batch/CHANGELOG.md:3]: This date needs to be within 30 days of the actual release date or the release pipeline will fail.

**Review by dpwatrous — CHANGES_REQUESTED** [review]

File: sdk/batch/Microsoft.Azure.Management.Batch/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,11 @@
 # Release History
 
+## 14.1.0 (2021-11-01)
+
+### Bug Fixes
+
+- Fixes a breaking type ambiguity in BatchAccountIdentity's constructor, introduced in #14.0.0.
```
Comment (dpwatrous) [inline sdk/batch/Microsoft.Azure.Management.Batch/CHANGELOG.md:7]: The # in front of the version number looks a bit odd here

**Review by dpwatrous — APPROVED** [review]

**Pull request merged**